src/chat_do.py

```
@@ -0,0 +1,406 @@
+    async def _load_history(self, classroom_id: str):
+        if self._history_loaded:
+            return
+        try:
+            raw_key = getattr(self.env, "CHAT_ENCRYPTION_KEY", None) or getattr(self.env, "ENCRYPTION_KEY", None)
+            enc_key = raw_key.strip() if isinstance(raw_key, str) and raw_key.strip() else ""
+            rows = await self.env.DB.prepare(
+                "SELECT id, user_id, display_name, content, created_at FROM chat_message WHERE classroom_id = ? ORDER BY created_at DESC LIMIT ?"
+            ).bind(classroom_id, self._MAX_BUFFER).all()
+            loaded = []
+            for row in (rows.results or []):
+                def _get(field):
```

Ruff (0.16.0)

[warning] Missing return type annotation for private function `_get`

(ANN202)